Radio.m

```diff
@@ -1445,14 +1577,62 @@ - (void) parseGpsToken: (NSScanner *) scan {
     
 }
 
-
+/**
+ * Parse Profile tokens
+ *     called on the GCD thread associated with the GCD tcpSocketQueue
+ *
+ *     format: <apiHandle>|profile <profileType> list=<value>^<value>^...<value>^
+ *                           OR
+ *     format: <apiHandle>|profile <profileType> current=<value>
+ *
+ *     scan is initially at scanLocation = 17, start of the <profileType>
+ *     "<apiHandle>|profile " has already been processed
+ */
 - (void) parseProfileToken: (NSScanner *) scan {
```

Comment:
What happens when a profile is deleted from the radio?  Does the radio tell you the profile (by name) has been removed or does it just send the entire new list?

Also, with objects like the list of profile names that can be accessed on multiple threads, its critical to guard changes to the Array property in a @synchronized block - like

@synchronized(self.txProfiles) {